test/note_test.rb

```diff
@@ -5,16 +5,56 @@ def setup
     @frequency   = 220.to_f
     @duration    = 2.0
     @sample_rate = 22050
+    @total_frames = (@duration * @sample_rate).to_i
+    @cycles_per_frame = @frequency / @sample_rate
+    @sine_wave_cycle = 2 * Math::PI * @cycles_per_frame
+    @samples = samples_helper
     @note = MusicTheory::Note.new(frequency: @frequency, duration: @duration)
+    @note_2 = MusicTheory::Note.new(frequency: @frequency, duration: @duration, distort: true)
   end
 
   def test_total_frames
-    frames = (@duration * @sample_rate).to_i
-    assert_equal frames, @note.total_frames
+    assert_equal @total_frames, @note.total_frames
   end
 
   def test_cycles_per_frame
-    cycles_per_frame = @frequency / @sample_rate
-    assert_equal cycles_per_frame, @note.cycles_per_frame
+    assert_equal @cycles_per_frame, @note.cycles_per_frame
+  end
+
+  def test_sine_wave_cycle
+    assert_equal @sine_wave_cycle, @note.sine_wave_cycle
+  end
+
+  def test_samples
+    assert_equal @samples, @note.samples
```

Comment:
I think a reasonable way to test this that doesn't involve copying over the original implementation would be to save the sample contents to a file that you can reference them in the test to assert that they're the same as the output of the known-good implementation.

Two options with saving executable ruby to disk... one of them is to use the [`Marshal`](http://ruby-doc.org/core-2.2.0/Marshal.html) class, which saves Ruby data as a byte stream and attempts to preserve as much of the context (i.e. interpreter state, Ruby version, etc) as it can. The other is to just `#to_s` your data structure and then just save it to a file. I think the later method is a fine choice here because the context doesn't really matter, you just care about the sample values.